pipeline.py

```diff
@@ -0,0 +1,72 @@
+import pmu,common
+from tabulate import tabulate
+import itertools,sys,re
+
+def pipeline_view(log,depths):
+  out_pipe=open(log.replace('csv','pipeline.log'),'w');
+  pipe_list=get_log(log)
+  event_list=[]
+  depth_list=[]
+  lines=[]
+  for i in depths:
+    event_list.append(depths[i][0])
+    depth_list.append(int(depths[i][1]))
+  issued_name=depths['issued'][0] 
+  issued_list=[i for i,item in enumerate(event_list) if re.search(issued_name,item)]
+  issued_index=issued_list[0]
+  lines=get_search_list(pipe_list,event_list[0])
+  rows=int(len(lines)/depth_list[0]) 
+  indices=[i+1 for i in range(max(depth_list))]
+  chunks=[[[0 for i in range(l)] for j in range(rows)] for k,l in zip(range(len(event_list)),depth_list)]
+  processed_chunks=[[[0 for i in range(l)] for j in range(rows)] for k,l in zip(range(len(event_list)),depth_list)]
+  percent_chunks=[[[0 for i in range(l)] for j in range(rows)] for k,l in zip(range(len(event_list)),depth_list)]
```

Comment:
I see here 3 lists are w/ the same definition, this is code duplication, you can write a small inner-function that returns this list and call it 3 times.

Reply:
fixed